cpp/cmake_modules/BuildUtils.cmake

```diff
@@ -370,7 +370,7 @@ function(ADD_ARROW_LIB LIB_NAME)
 
     target_link_libraries(${LIB_NAME}_shared
                           LINK_PUBLIC
-                          "$<BUILD_INTERFACE:${ARG_SHARED_LINK_LIBS}>"
+                          ${ARG_SHARED_LINK_LIBS}
```

Comment:
TODO: This works on my machine, but there's probably a reason `BUILD_INTERFACE` was there.

Reply:
It exports too much libraries to `ArrowTargets.cmake`. See also: 8ab1493c810ae354ce085c2c2052676f349b168a

What build error was solved by this change?
Could you show build log for this?

Reply:
I can show a full log later, but I've described what I was seeing in this Jira comment: https://issues.apache.org/jira/browse/ARROW-13470?focusedCommentId=17480335

Reply:
Build error:

```
[build] [381/559] Linking CXX shared library debug\arrow_python.dll
[build] FAILED: debug/arrow_python.dll debug/arrow_python.lib 
[build] cmd.exe /C "cd . && "C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe" -E vs_link_dll --intdir=src\arrow\python\CMakeFiles\arrow_python_shared.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MICROS~4\2022\COMMUN~1\VC\Tools\MSVC\1430~1.307\bin\Hostx64\x64\link.exe /nologo src\arrow\python\CMakeFiles\arrow_python_shared.dir\arrow_to_pandas.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\benchmark.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\common.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\datetime.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\decimal.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\deserialize.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\extension_type.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\helpers.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\inference.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\init.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\io.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\ipc.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\numpy_convert.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\numpy_to_arrow.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\python_to_arrow.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\pyarrow.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\serialize.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\csv.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\filesystem.cc.obj  /out:debug\arrow_python.dll /implib:debug\arrow_python.lib /pdb:debug\arrow_python.pdb /dll /version:700.0 /machine:x64  /NODEFAULTLIB:LIBCMT /debug /INCREMENTAL -Wl,--version-script=C:/Users/voltron/arrow/cpp/src/arrow/symbols.map  debug\arrow.lib  optimized.lib  C:\Users\voltron\AppData\Local\Programs\Python\Python310\libs\python310.lib  debug.lib  C:\Users\voltron\AppData\Local\Programs\Python\Python310\libs\python310_d.lib  ..\..\vcpkg_installed\x64-windows\lib\brotlienc.lib  ..\..\vcpkg_installed\x64-windows\lib\brotlidec.lib  ..\..\vcpkg_installed\x64-windows\lib\brotlicommon.lib  ..\..\vcpkg_installed\x64-windows\debug\lib\utf8proc.lib  ..\..\vcpkg_installed\x64-windows\debug\lib\re2.lib  mimalloc_ep\src\mimalloc_ep\lib\mimalloc-1.7\mimalloc-static-debug.lib  ws2_32.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib  && cmd.exe /C "cd /D C:\Users\voltron\arrow\cpp\build\user-python-debug-windows\src\arrow\python && C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -noprofile -executionpolicy Bypass -file C:/Users/voltron/arrow/vcpkg/scripts/buildsystems/msbuild/applocal.ps1 -targetBinary C:/Users/voltron/arrow/cpp/build/user-python-debug-windows/debug/arrow_python.dll -installedDir C:/Users/voltron/arrow/cpp/vcpkg_installed/x64-windows/debug/bin -OutVariable out""
[build] LINK Pass 1: command "C:\PROGRA~1\MICROS~4\2022\COMMUN~1\VC\Tools\MSVC\1430~1.307\bin\Hostx64\x64\link.exe /nologo src\arrow\python\CMakeFiles\arrow_python_shared.dir\arrow_to_pandas.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\benchmark.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\common.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\datetime.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\decimal.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\deserialize.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\extension_type.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\helpers.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\inference.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\init.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\io.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\ipc.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\numpy_convert.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\numpy_to_arrow.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\python_to_arrow.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\pyarrow.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\serialize.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\csv.cc.obj src\arrow\python\CMakeFiles\arrow_python_shared.dir\filesystem.cc.obj /out:debug\arrow_python.dll /implib:debug\arrow_python.lib /pdb:debug\arrow_python.pdb /dll /version:700.0 /machine:x64 /NODEFAULTLIB:LIBCMT /debug /INCREMENTAL -Wl,--version-script=C:/Users/voltron/arrow/cpp/src/arrow/symbols.map debug\arrow.lib optimized.lib C:\Users\voltron\AppData\Local\Programs\Python\Python310\libs\python310.lib debug.lib C:\Users\voltron\AppData\Local\Programs\Python\Python310\libs\python310_d.lib ..\..\vcpkg_installed\x64-windows\lib\brotlienc.lib ..\..\vcpkg_installed\x64-windows\lib\brotlidec.lib ..\..\vcpkg_installed\x64-windows\lib\brotlicommon.lib ..\..\vcpkg_installed\x64-windows\debug\lib\utf8proc.lib ..\..\vcpkg_installed\x64-windows\debug\lib\re2.lib mimalloc_ep\src\mimalloc_ep\lib\mimalloc-1.7\mimalloc-static-debug.lib ws2_32.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:src\arrow\python\CMakeFiles\arrow_python_shared.dir/intermediate.manifest src\arrow\python\CMakeFiles\arrow_python_shared.dir/manifest.res" failed (exit code 1104) with the following output:
[build] LINK : warning LNK4044: unrecognized option '/Wl,--version-script=C:/Users/voltron/arrow/cpp/src/arrow/symbols.map'; ignored
[build] LINK : fatal error LNK1104: cannot open file 'optimized.lib'
```